util/config/compileline

```diff
@@ -1,154 +1,186 @@
-    chomp $maino;
-    if( ! -e $maino ) {
-      print STDERR "Missing clang-included runtime build for chpl --llvm\n";
-      exit 1;
-    }
-  } elsif( $arg eq "--home" ) {
-    print $chpl_home_dir . "\n";
-  } elsif( $arg eq "--make" ) {
-    print $make . "\n";
-  } elsif( $arg eq "--compile" ) {
-    mysystem("$make -f $chpl_home_dir/runtime/etc/Makefile.include printcompileline");
-  } elsif( $arg eq "--cflags" ) {
-    mysystem("$make -f $chpl_home_dir/runtime/etc/Makefile.include printcflags");
-  } elsif ( $arg eq "--includes-and-defines" ) {
-    mysystem("$make -f $chpl_home_dir/runtime/etc/Makefile.include printincludesanddefines");
-  } elsif( $arg eq "--libraries" ) {
-    mysystem("$make -f $chpl_home_dir/runtime/etc/Makefile.include printlibraries");
-  } elsif( $arg eq "--main.o" ) {
-    mysystem("$make -f $chpl_home_dir/runtime/etc/Makefile.include printmaino");
-  } elsif( $arg eq "--llvm-install-dir" ) {
-    mysystem("$make -f $chpl_home_dir/runtime/etc/Makefile.include printllvminstall");
-  } elsif( $arg eq "--clang" ) {
-    mysystem("$make -f $chpl_home_dir/runtime/etc/Makefile.include printclangcc printclangcxx");
-  } elsif( $arg eq "--clang-sysroot-arguments" ) {
-    $llvminstall=`$make -f $chpl_home_dir/runtime/etc/Makefile.include printllvminstall`;
-    chomp $llvminstall;
-    open(ARGS, "<", "$llvminstall/configured-clang-sysroot-arguments") or die "Could not open $llvminstall/configured-clang-sysroot-arguments: $!";
-    while(<ARGS>) {
-      chomp;
-      print " $_";
-    }
-    print "\n";
-    close(ARGS);
-  } elsif( $arg =~ /[^-=][^=]*=[^=]+/ ) {
-    ($key, $val) = split('=', $arg, 2);
-    $ENV{$key} = $val;
-  } else {
-    print "Unknown argument $arg\n";
-    print "Usage: compileline [VAR=VALUE...] [options...]\n";
-    print "[VAR=VALUE] sets any environment variable\n";
-    print "[options] include\n";
-    print " --llvm specify that the rest of the arguments are for a compile\n";
-    print "        working with the LLVM backend. This argument causes\n";
-    print "        this script to set CHPL_TARGET_COMPILER=clang-included\n";
-    print " --home  print out CHPL_HOME\n";
-    print " --make  print out the make command\n";
-    print " --compile  print out the command to compile a file including\n";
-    print "            Chapel runtime headers\n";
-    print " --includes-and-defines print out any -D and -I options to\n";
-    print "                        include the Chapel runtime\n";
-    print " --libraries print out any -L and -l options to\n";
-    print "             link with the Chapel runtime\n";
-    print " --main.o print out the path to the main.o file\n";
-    print " --clang print out the path to clang and clang++\n";
-    print " --clang-sysroot-arguments print out any saved clang arguments\n";
-    print "                           that specify the system root\n";
-  }
-}
+  ret = ""
+  output = subprocess.check_output(cmd, shell=True)
```

Comment:
I haven't taken a real look at this, but just a high level comment: please make sure this works with python 2.6, 2.7, and 3.x. check_output is new in 2.7 so you can't use it, but there may be other things that need to be addressed too.

You can probably import util/chplenv/utils.py and re-use run_command (a check_output replacement)